extensions/2.0/Khronos/KHR_meshopt_compression/README.md

```diff
@@ -0,0 +1,654 @@
+
+This filter is only valid if `byteStride` is a multiple of 4.
+
+The input to the filter is a sequence of 32-bit little endian integers, with the most significant 8 bits specifying a (signed) exponent value, and the remaining 24 bits specifying a (signed) mantissa value. The integers are stored in two-complement format.
+
+The result of the filter is 2^e * m:
+
+```
+float32_t decode(int32_t input) {
+	int32_t e = input >> 24;
+	int32_t m = (input << 8) >> 8;
+	return pow(2.0, e) * m;
+}
+```
+
+The valid range of `e` is [-100, +100], which facilitates performant implementations. Decoding out of range values results in unspecified behavior, and encoders are expected to clamp `e` to the valid range.
+
+## Filter 4: color
+
+Color filter allows to encode color data using YCoCg color space transformation, which results in better compression for typical color data by exploiting correlation between color channels.
+
+This filter is only valid if `byteStride` is 4 or 8. When `byteStride` is 4, then the input and output of this filter are four 8-bit components, and when `byteStride` is 8, the input and output of this filter are four 16-bit components.
+
+The input to the filter is four 8-bit or 16-bit components, where the first component stores the Y (luma) value as a K-bit unsigned integer, the second and third components store Co/Cg (chrominance) values as K-bit signed integers, and the fourth component stores the alpha value as a K-1-bit unsigned integer with the bit K set to 1. 1 <= K <= 16, signed integers are stored in two's complement format.
+
+The transformation uses YCoCg encoding; reconstruction of RGB values can be performed in integer space or in floating point space, depending on the implementation. The encoder must guarantee that RGB can be reconstructed using K-bit integer math without overflow or underflow.
+
+The alpha component uses K-1 bits for the alpha value with the high bit set to 1; note that K can be smaller than the bit depth. This allows decoder to recover K and decode the color and alpha values correctly.
+
+The output of the filter is four decoded color components (R, G, B, A), stored as 8-bit or 16-bit normalized integers.
+
+```
+void decode(intN_t input[4], intN_t output[4]) {
+	// recover scale from alpha high bit
+	int as = (1 << (findMSB(input[3]) + 1)) - 1;
+
+	// convert to RGB in fixed point
+	int y = input[0], co = input[1], cg = input[2];
+
+	int r = y + co - cg;
+	int g = y + cg;
+	int b = y - co - cg;
+
+	// expand alpha by one bit to match other components, replicating last bit
+	int a = input[3] & (as >> 1);
+	a = (a << 1) | (a & 1);
+
+	// compute scaling factor
+	float ss = INTN_MAX / float(as);
+
+	// rounded float->int
+	output[0] = int(float(r) * ss + 0.5f);
+	output[1] = int(float(g) * ss + 0.5f);
+	output[2] = int(float(b) * ss + 0.5f);
+	output[3] = int(float(a) * ss + 0.5f);
```

Comment:
This expression does not strictly match any definition of the `round` operation. If the multiplication result is `0.49999997f`, the rounded value would be 1. Even if that exact case is not technically reachable given the limited ranges of input integer values, it would still be better to not introduce such discrepancies on the spec level.

I'd suggest using `round` here instead and making sure that it's precisely defined in the spec (see above).